driver-core/src/main/com/mongodb/internal/session/BaseClientSessionImpl.java

```diff
@@ -182,8 +183,7 @@ private BsonTimestamp greaterOf(@Nullable final BsonTimestamp newOperationTime)
 
     @Override
     public void close() {
-        if (!closed) {
-            closed = true;
+        if (closed.compareAndSet(false, true)) {
```

Comment:
[optional]
It will be helpful to have a comment here explaining why synchronization exists in a class, instances of which must not be used concurrently. Such a comment may address

- why concurrency is relevant for instances of this class at all
- what specific problem is solved here
- why problems that concurrency may cause in other places of this class are disregarded